module/lua/llex.c

```diff
@@ -431,10 +431,12 @@ static int llex (LexState *ls, SemInfo *seminfo) {
         if (sep >= 0) {
           read_long_string(ls, seminfo, sep);
           return TK_STRING;
-        }
-        else if (sep == -1) return '[';
-        else lexerror(ls, "invalid long string delimiter", TK_STRING);
+        } else if (sep == -1)
+		return '[';
+        else
+		lexerror(ls, "invalid long string delimiter", TK_STRING);
```

Comment:
Could we move the `break` in to the `else` block.  It looks like the compiler warning is the result of it not recognizing that `lexerror` is fatal (noreturn).  Maybe,

```c
        } else if (sep == -1) { 
          return '[';
        } else {
          lexerror(ls, "invalid long string delimiter", TK_STRING);
          break;
        }
```